File: 2024/ruby/01.rb
# frozen_string_literal: true

inputs = File.read("../input/01#{ARGV[0]}.input").split("\n")

pairs = inputs.map(&:split).map { |i| i.map(&:to_i) }

column_one = pairs.map { |i| i[0] }.sort
column_two = pairs.map { |i| i[1] }.sort

sorted_pairs = column_one.zip(column_two)

puts sorted_pairs.map { |x, y| (x - y).abs }.sum

File: 2024/python/01.py
inputs = open("../input/01.input").read().splitlines()

import pdb; pdb.set_trace()
pairs = [
    [int(i) for i in i.split()]
    for i in inputs
]

columns = list(map(list, zip(*pairs)))
sorted_pairs = list(zip(sorted(columns[0]), sorted(columns[1])))

total = 0
for pair in sorted_pairs:
    total += abs(pair[0] - pair[1])

print(total)
